### PR TITLE
Inserting fragments should not remove image nodes

### PR DESCRIPTION
**Pull request by covertcj**
* insertTreeFragmentIntoRange now checks for image nodes in addition to
the text content when removing empty nodes

### CODE REVIEW

File: build/squire-raw.js [hunk 1]
```diff
@@ -889,6 +889,28 @@ var deleteContentsOfRange = function ( range, root ) {
     return frag;
 };
 
+var isNodeEmpty = function ( node ) {
+    if (!node) {
+        return false;
+    }
+
+    // check for image nodes
+    if ( node.nodeType === Node.ELEMENT_NODE && node.nodeName === 'IMG' ) {
```
Comment (covertcj) [inline build/squire-raw.js]: Ideally the next if statement with the query selector would be able to handle the case in which the top level node is an image as well.  Does anyone know of a way to combine these two ifs?  Maybe a way to make querySelector check itself as well?

Comment (akauffmanGG) [reply build/squire-raw.js]: I couldn't find anything in the Selectors API similar to jquery's addBack() either. However I would just combine the ifs like  `if ( node.nodeName === 'IMG' || node.querySelectory( 'img' ) )`
and then just drop the nodeType check.

File: build/squire-raw.js [hunk 2]
```diff
@@ -889,6 +889,28 @@ var deleteContentsOfRange = function ( range, root ) {
     return frag;
 };
 
+var isNodeEmpty = function ( node ) {
+    if (!node) {
```
Comment (akauffmanGG) [inline build/squire-raw.js]: spacing

File: build/squire-raw.js [hunk 3]
```diff
@@ -889,6 +889,28 @@ var deleteContentsOfRange = function ( range, root ) {
...
+        return false;
+    }
+
+    // check for image nodes
+    if ( node.nodeType === Node.ELEMENT_NODE && node.nodeName === 'IMG' ) {
+        return false;
+    }
+
+    if ( node.querySelector('img') ) {
```
Comment (akauffmanGG) [inline build/squire-raw.js]: spacing

**Pull request merged**